Dockerfile

```diff
@@ -93,7 +80,6 @@ RUN --mount=type=cache,sharing=locked,id=moby-frozen-images-aptlib,target=/var/l
 COPY contrib/download-frozen-image-v2.sh /
 ARG TARGETARCH
 RUN /download-frozen-image-v2.sh /build \
-        buildpack-deps:buster@sha256:d0abb4b1e5c664828b93e8b6ac84d10bce45ee469999bef88304be04a2709491 \
```

Comment:
This removes the buildpack-deps frozen image, but doesn't update the code that uses it (see https://github.com/moby/moby/commit/f68260ba228349d0634235e7c314b4ed88083c62 which made the same changes, but also updated other parts);

have you also tried if cherry-picking the relevant commits from https://github.com/moby/moby/pull/42763 works? (there likely will be some conflicts though)

Reply:
Had not tried cherry-picking the commits. Now, I have updated the PR with all the required changes.